src/h5web/visualizations/line/DataCurve.tsx

```diff
@@ -19,45 +22,82 @@ function DataCurve(props: Props): ReactElement {
   const {
     abscissas,
     ordinates,
+    errors,
+    showErrors,
     color = DEFAULT_COLOR,
     curveType = CurveType.LineOnly,
   } = props;
 
   const { camera } = useThree();
   const { abscissaScale, ordinateScale } = useCanvasScales();
 
-  const dataGeometry = useMemo(() => {
-    const points = ordinates.map((val, index) => {
-      const ordinate = ordinateScale(val);
-      return new Vector3(
-        abscissaScale(abscissas[index]),
-        // This is to avoid a three.js warning when ordinateScale(val) is Infinity
-        Number.isFinite(ordinate) ? ordinate : 0,
-        // Move NaN/Infinity out of the camera FOV (negative val for logScale).
-        // This allows to have only curve segments for the positive values
-        Number.isFinite(ordinate) ? 0 : camera.far
-      );
+  const points = useMemo(() => {
+    const dataPoints: Vector3[] = [];
+    const errorBarSegments: Vector3[] = [];
+    const errorCapPoints: Vector3[] = [];
+
+    ordinates.forEach((val, index) => {
+      const x = abscissaScale(abscissas[index]);
+      const y = ordinateScale(val);
+
+      // Move NaN/Infinity out of the camera FOV (negative val for logScale).
+      // This allows to have only curve segments for the positive values
+      const finiteData = Number.isFinite(x) && Number.isFinite(y);
+      const dataVector = finiteData
+        ? // Set x,y to 0 to avoid a three.js warning when one is Infinity
+          new Vector3(x, y, 0)
+        : // Move NaN/Infinity out of the camera FOV (negative val for logScale).
+          // This allows to have only curve segments for the positive values
+          new Vector3(0, 0, camera.far);
+
+      dataPoints.push(dataVector);
+      if (!errors || !errors[index] || !finiteData) {
```

Comment:
Maybe add a line `const error = errors && errors[index]` and then check `!error || !isFiniteData`? This way you don't need to reaccess the `error` value with the `index` below.